nuxeo-core-binarymanager-s3/src/test/java/org/nuxeo/ecm/core/storage/sql/TestCloudFrontRedirect.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * (C) Copyright 2011-2017 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Arnaud Kervern
+ */
+package org.nuxeo.ecm.core.storage.sql;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assume.assumeTrue;
+import static org.nuxeo.ecm.core.storage.sql.CloudFrontBinaryManager.DISTRIB_DOMAIN_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.CloudFrontBinaryManager.PRIVATE_KEY_ID_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.CloudFrontBinaryManager.PRIVATE_KEY_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.AWS_ID_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.AWS_SECRET_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.BUCKET_NAME_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.BUCKET_PREFIX_PROPERTY;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+
+/**
+ * Test ignored as it requires lots of file and tokens. Manually executed
+ */
+@Ignore
+@RunWith(FeaturesRunner.class)
+@Features(RuntimeFeature.class)
+public class TestCloudFrontRedirect extends AbstractS3BinaryTest<CloudFrontBinaryManager> {
+    protected static Map<String, String> PROPERTIES = Collections.emptyMap();
+
+    @BeforeClass
+    public static void beforeClass() {
```

Comment:
This looks like a copy/paste of `TestS3BinaryManager`, but without the ability to use System properties.
It looks like there is a space for factorization up to `AbstractS3BinaryTest`.